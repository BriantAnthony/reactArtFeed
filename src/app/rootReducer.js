import { combineReducers } from 'redux';
import userReducer from '../user/userReducer';
import cardReducer from '../cardFeed/cardReducer';

const rootReducer = combineReducers({
  app: userReducer,
  artCard: cardReducer
})

export default rootReducer;