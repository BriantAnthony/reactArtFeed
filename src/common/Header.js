import React, { Component } from 'react';
import { connect } from 'react-redux';
import { userLoginThunk, userLogoutThunk } from '../user/UserActions';
import AppBar from 'material-ui/AppBar';
import FlatButton from 'material-ui/FlatButton';

class Header extends Component {

  handleTouchTap() {
    if(!this.props.isAuthenticated){
      this.props.dispatch(userLoginThunk({email: 'briantacampbell@gmail.com', password: 'demo'}));
    } else {
      this.props.dispatch(userLogoutThunk());
    }
  }

  render() {
    //console.log('props: ', this.props);
    const authLabel = () => {
      if(this.props.isAuthenticated){
        return 'LOGOUT'
      } else {
        return 'LOGIN'
      }
    }
    return (
      <AppBar 
        title="Art Safari"
        iconElementRight={<FlatButton label={authLabel()} />}
        onRightIconButtonTouchTap={this.handleTouchTap.bind(this)}
      />
    );
  }
}

const mapStateToProps = (state) => {
  //console.log('state: ', state);
  return {
    user : state.app.user,
    isAuthenticated: state.app.isAuthenticated
  }
};

export default connect(mapStateToProps)(Header);