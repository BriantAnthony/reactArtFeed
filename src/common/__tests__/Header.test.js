import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../../app/rootReducer';
import Header from '../Header';
import renderer from 'react-test-renderer';

jest.mock('material-ui/AppBar', () => 'AppBar');
const store = createStore(rootReducer);

it('renders without crashing', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Header />
    </Provider>
  ).toJSON();
  //console.log('tree: ', tree);
  expect(tree).toMatchSnapshot();
});
