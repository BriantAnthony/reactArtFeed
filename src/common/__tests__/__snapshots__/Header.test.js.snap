exports[`test renders without crashing 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }>
  <button
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button">
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}>
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#ffffff",
            "display": "inline-block",
            "fill": "#ffffff",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24">
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
      </svg>
    </div>
  </button>
  <h1
    onTouchTap={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }>
    Art Safari
  </h1>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": -16,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }>
    <button
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "#ffffff",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "marginTop": 7,
          "minWidth": 88,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button">
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }>
          LOGIN
        </span>
      </div>
    </button>
  </div>
</div>
`;
